src/SiteConfig/ConfigBuilder.php

```diff
@@ -29,22 +28,9 @@ class ConfigBuilder
         'div',
     ];
 
-    /**
-     * @param array $config
-     */
-    public function __construct($config = [], LoggerInterface $logger = null)
+    public function __construct(array $config = [], LoggerInterface $logger = null)
```

Comment:
I agree it could allow to better check for sub-configuration but I found it more verbose. You'll have to _use_ object config for each level. I think it'll add complexity where array is really simple to use.